src/main/java/cz/brmlab/yodaqa/pipeline/structured/StructuredPrimarySearch.java

```diff
@@ -135,13 +142,22 @@ protected void propertyToAnswer(JCas jcas, PropertyValue property,
 		jcas.setDocumentLanguage("en"); // XXX
 
 		String title = property.getObject() + " " + property.getProperty();
+		AnswerSourceStructured asf;// new AnswerSourceStructured("structured",property.getValRes(),property.getObject());
+		AnsweringProperty ap; //= new AnsweringProperty(SnippetIDGenerator.getInstance().generateID(), asf.getSourceID(), property.getProperty());
+		int sourceID = getSourceID(property.getValRes(),property.getObject(),questionView);
+
+		ap = new AnsweringProperty(SnippetIDGenerator.getInstance().generateID(), sourceID, property.getProperty());
+
+		QuestionDashboard.getInstance().addSnippet(ap.getSnippetID(),ap);
 
 		ResultInfo ri = new ResultInfo(jcas);
 		ri.setDocumentTitle(title);
 		ri.setSource(sourceName);
 		ri.setRelevance(1.0);
 		ri.setIsLast(isLast);
+		ri.setSourceID(sourceID);
 		ri.setOrigin(this.getClass().getCanonicalName());
+
```

Comment:
The idea here is that ri setup is a single compact block of code, from instantiation to addtoIndexes.